### ISSUE TITLE
AttributeError: 'SimilarityEncoder' object has no attribute 'drop_idx_'

### ISSUE THREAD

**Issue by LilianBoulard** [dirty-cat/dirty_cat#168]
Hello

While working today with the SimilarityEncoder, I found this error :
```
/usr/bin/python3.8 /home/lboulard/.config/JetBrains/PyCharmCE2021.1/scratches/Issue_with_similarity_encoder.py
Traceback (most recent call last):
  File "Issue_with_similarity_encoder.py", line 11, in <module>
    feature_names = enc.get_feature_names()
  File "~/.local/lib/python3.8/site-packages/sklearn/preprocessing/_encoders.py", line 590, in get_feature_names
    if self.drop_idx_ is not None and self.drop_idx_[i] is not None:
AttributeError: 'SimilarityEncoder' object has no attribute 'drop_idx_'
```

Code to reproduce : 
```
from dirty_cat import SimilarityEncoder
from dirty_cat.datasets import fetch_employee_salaries

employee_salaries = fetch_employee_salaries()

# Get only one column
X = employee_salaries['data']['employee_position_title'].to_numpy().reshape(-1, 1)

enc = SimilarityEncoder()
enc.fit(X)
feature_names = enc.get_feature_names()
print(feature_names)
```

While looking at the code, I found that, in `sklearn.OneHotEncoder` - that SimilarityEncoder inherits - there is this line in the `fit` method :

[`self.drop_idx_ = self._compute_drop_idx()`](https://github.com/scikit-learn/scikit-learn/blob/2beed55847ee70d363bdbfe14ee4401438fba057/sklearn/preprocessing/_encoders.py#L425)

Which is missing from `SimilarityEncoder.fit()`.

**Easy fix: add the exact same line in the method, which fixes the issue.**

Disclaimer: I haven't looked very much in depth into what `OneHotEncoder._compute_drop_idx()` does. 
I'll do when I have a little more time :)  
So... might not be the best fix (tough in my case it works).

Please let me know if this looks good to you, I have a PR ready.

**Comment by GaelVaroquaux** [comment]
Seems like a good fix.

Please send a PR with the fix and a test. Thanks a lot!!

**Comment by GaelVaroquaux** [comment]
Fixed by #169
